qiskit/opflow/gradients/circuit_gradients/circuit_gradient.py

```diff
@@ -66,3 +68,28 @@ def convert(
             ValueError: If ``params`` contains a parameter not present in ``operator``.
         """
         raise NotImplementedError
+
+    @staticmethod
+    def _unroll_to_supported_operations(
+        circuit: QuantumCircuit, supported_gates: Set[str]
+    ) -> QuantumCircuit:
+        """Unroll the given circuit into a gate set for which the gradients may be computed.
+
+        Args:
+            circuit: Quantum circuit to be unrolled into supported operations.
+
+        Returns:
+            Quantum circuit which is unrolled into supported operations.
+
+        """
+        unique_ops = set(circuit.count_ops().keys())
+        if not unique_ops.issubset(supported_gates):
+            try:
+                unroller = Unroller(list(supported_gates))
+                circuit = unroller(circuit)
+            except Exception:
+                raise Exception(
+                    f"Could not unroll the circuit provided {circuit} into supported gates "
+                    f"{supported_gates}."
+                )
```

Comment:
This should catch a more specific exception that everything, raise a more specific exception than the base exception, and also raise from the original exception.

Reply:
Yes, forgot to revisit it. Fixed now.